plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -397,4 +398,19 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
 
         return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, targetPath, propertiesBuilder.build());
     }
+
+    public static long getSnapshotIdAsOfTime(Table table, long timestampMillis)
+    {
+        Long snapshotId = null;
+        for (HistoryEntry logEntry : table.history()) {
+            if (logEntry.timestampMillis() <= timestampMillis) {
+                snapshotId = logEntry.snapshotId();
+            }
+        }
+
+        if (snapshotId == null) {
+            throw new TrinoException(GENERIC_USER_ERROR, "Cannot find a snapshot older than epoch time " + timestampMillis);
```

Comment:
More of a semantic question for the `AS OF` syntax, but should this fail or should it return an empty set of data?

Reply:
I think it should fail, returning an empty set of data should mean the table snapshot at that point of time actually contains no data.

Reply:
And this will also match the Spark SQL behavior in Iceberg.

Reply:
I agree with @jackye1995 that the query should fail instead of returning an empty set of data.